packages/common/typings/src/tuple.ts

```diff
@@ -0,0 +1,3 @@
+export type Tuple<ItemType, Length extends number = 1, Base extends unknown[] = []> = Base['length'] extends Length
```

Comment:
what do you think about this? https://github.com/microsoft/TypeScript/issues/26223#issuecomment-792380120
we really get an error if we specify a length greater than 45 in the recursive approach

Reply:
@Schirbak what do you think about?

Reply:
#691